source/apiVolontaria/volunteer/tests/tests_admin_Cycle.py

```diff
@@ -0,0 +1,134 @@
+
+    def setUp(self):
+        self.create_user()
+
+        self.random_country = Country.objects.create(
+            name="random country",
+            iso_code="RC",
+        )
+        self.random_state_province = StateProvince.objects.create(
+            name="random state",
+            iso_code="RS",
+            country=self.random_country,
+        )
+        self.address = Address.objects.create(
+            address_line1='random address 1',
+            postal_code='RAN DOM',
+            city='random city',
+            state_province=self.random_state_province,
+            country=self.random_country,
+        )
+        self.cell = Cell.objects.create(
+            name="my cell",
+            address=self.address,
+        )
+        self.cycle = Cycle.objects.create(
+            name="my cycle",
+        )
+        self.task_type = TaskType.objects.create(
+            name="my tasktype",
+        )
+
+        start_date = timezone.now() - timezone.timedelta(
+            minutes=100,
+        )
+        end_date = start_date + timezone.timedelta(
+            minutes=50,
+        )
+
+        self.cycle_inactive = Cycle.objects.create(
+            name="my cycle",
+            start_date=start_date,
+            end_date=end_date,
+        )
+
+        # Some date INSIDE the cycle range
+        start_date = start_date + timezone.timedelta(
+            minutes=1,
+        )
+        end_date = end_date - timezone.timedelta(
+            minutes=1,
+        )
+
+        self.event = Event.objects.create(
+            cell=self.cell,
+            cycle=self.cycle,
+            task_type=self.task_type,
+            start_date=start_date,
+            end_date=end_date,
+        )
+
+        self.participation = Participation(event=self.event, user=self.user)
+
+        self.site = AdminSite()
+        self.admin = CycleAdmin(Cycle, self.site)
+
+        self.profile = Profile.objects.filter(
+            user__pk=self.participation.user.pk
+        ).first()
+
+        self.client = Client()
+        self.client.login(username=self.username, password=self.password)
+
+    def test_admin_access(self):
+        admin_pages = [
+            "/admin/volunteer/",
+            "/admin/volunteer/cycle/",
+            ]
+
+        for page in admin_pages:
+            resp = self.client.get(page)
+            self.assertEqual(resp.status_code, 200)
+            self.assertIn("<!DOCTYPE html", str(resp.content))
+
+    def test_admin_participations(self):
+        self.assertNotEqual(self.admin, None)
+
+    def test_admin_cycle_participation_export(self):
+        change_url = urlresolvers.reverse('admin:volunteer_cycle_changelist')
+
+        data = {'action': 'cycle_admin_generate_participation_report',
+                ACTION_CHECKBOX_NAME: [self.cycle.pk, ]}
+
+        response = self.client.post(change_url, data)
+
+        print(response.status_code)
+
+        self.assertEquals(response.status_code, 200)
```

Comment:
We definitively want to test the content of the file and not just the status code!

Reply:
Yes added the actual text